PCF-ADR-0008-ray-cluster-authentication.md

```diff
@@ -0,0 +1,89 @@
+# Project CodeFlare - RayCluster Authentication
+|                |                                                                                  |
+| -------------- | -------------------------------------------------------------------------------- |
+| Date           | 12/19/2023                                                                       |
+| Scope          |                                                                                  |
+| Status         | provisional                                                                      |
+| Authors        | [Kevin Postlethwait](@KPostOffice)                                               |
+| Supersedes     | N/A                                                                              |
+| Superseded by: | N/A                                                                              |
+| Issues         |                                                                                  |
+| Other docs:    | none                                                                             |
+
+## What
+
+Enabling OpenShift OAuth as an identity provider and authorization mechanism for KubeRay RayClusters.
+
+## Why
+
+Currently there is no authentication by default enabled in front of the Ray Dashboard and we would like to offer a
+unified login experience for the whole of ODH by way of OpenShift OAuth
+
+## Goals
+
+* Protect the dashboard from unauthenticated access by way of an OAuth flow
+* Protect the API endpoint from unauthenticated access by way of bearer tokens
+* Enable HTTPS by default for accessing the RayCluster API and Dashboard
```

Comment:
Should we mention the "interactive session" use case, i.e., using the Ray CLI locally connected to a secured Ray cluster deployed on OpenShift?